staging/src/k8s.io/endpointslice/util/endpoint/controller_utils.go

```diff
@@ -302,3 +302,75 @@ func stringPtrChanged(ptr1, ptr2 *string) bool {
 func EndpointSubsetsEqualIgnoreResourceVersion(subsets1, subsets2 []v1.EndpointSubset) bool {
 	return semanticIgnoreResourceVersion.DeepEqual(subsets1, subsets2)
 }
+
+// DeepHashObject writes specified object to hash using the spew library
+// which follows pointers and prints actual values of the nested objects
+// ensuring the hash does not change when a pointer changes.
+func deepHashObject(hasher hash.Hash, objectToWrite interface{}) {
+	hasher.Reset()
+	printer := spew.ConfigState{
+		Indent:         " ",
+		SortKeys:       true,
+		DisableMethods: true,
+		SpewKeys:       true,
+	}
+	printer.Fprintf(hasher, "%#v", objectToWrite)
+}
+
+func IsPodReady(pod *v1.Pod) bool {
```

Comment:
If we flatten the relevant utils into `k8s.io/endpointslice`, then this can be made private.